lib/src/main/java/com/diffplug/spotless/TimedLogger.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2023 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless;
+
+import static com.diffplug.spotless.LazyArgLogger.lazy;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import javax.annotation.Nonnull;
+
+import org.slf4j.Logger;
+
+/**
+ * A logger that logs the time it took to execute a block of code.
+ */
+public class TimedLogger {
```

Comment:
I would prefer this to be package-private, which would mean moving it to the package where it is actually used.

Reply:
I kind of made that public on purpose thinking that this might be useful in other cases, too. If you prefer, I'll change it to package private no problem 😄